### PR TITLE
change download and images view purge path adapter registration to IDexterityContent

### LINKED ISSUE #136: @@download and @@images are never purged
There is a `IPurgePaths` adapter registered that should purge the `@@downloads` and `@@images` view on content editing but it is registered for `IDexteritySchema` which is wrong. [see the registration here](https://github.com/plone/plone.app.caching/blob/master/plone/app/caching/purge.py#L174-L176)

It should be registered for `IDexterityContent` (or `IDexterityItem` and `IDexterityContainer` I don't mind) in order to correctly purge those URLs.

If we go 7 years back on time to the commit that introduced the adapter [see here](https://github.com/plone/plone.app.caching/commit/6e725b258d85591f9db579253dc6b233708918a4), the commit message says *Purging all image scale paths and file paths in custom dexterity content types*, which was the intention but never happened.

The point here is that when running the tests the created content type provides `IDexteritySchema` but when creating the object in Plone the created object does not provide the said interface.

### PR DESCRIPTION
**Pull request by erral**
Fixes #136

### CODE REVIEW

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

**Review by davisagli — COMMENTED** [review]
Is there any test for this functionality?

**Comment by erral** [comment]
> Is there any test for this functionality?

Yes there is. The previous test keeps working.

I don't know why but when running tests the created object provides the IDexteritySchema but when creating a content object in Plone it doesn't. That's why the previous tests keep working. See here:
https://github.com/plone/plone.app.caching/blob/master/plone/app/caching/tests/test_purge.py#L330

File: news/136.bugfix [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Change PurgePath adapter registration to IDexterityContent
```
Comment (davisagli) [inline news/136.bugfix:1]:
```suggestion
Fix purging of image scale paths for Dexterity content.
```

**Review by davisagli — APPROVED** [review]

**Pull request merged**